### DISCUSSION TITLE
Is it possible to add similar search support for misspelled words ?

### DISCUSSION THREAD

**Discussion by ghevge** [pgroonga/pgroonga#320] [Q&A]
Is it possible to add similar search support for misspelled words ?
eg: I would like to get a similar result when I issue:

SELECT product_id FROM search WHERE text &@* 'vineger';

as when I issue:

SELECT product_id FROM search WHERE text &@* 'vinegar';

note the typo in : vineger

**Comment by komainu8** [comment]
How about use fuzzy search if you want to search even if there is a bit of typo?
How to use the fuzzy search as below.

```sql
CREATE TABLE search (
  product_id integer,
  name text
);

CREATE INDEX name_search
          ON search
       USING pgroonga(name)
        WITH (tokenizer='');

INSERT INTO search VALUES(1, 'vineger');

SELECT *
  FROM search
 WHERE name &` ('fuzzy_search(name, ' || pgroonga_escape('vinegar') || ', {"with_transposition": true, "max_distance": 5})');
 product_id |  name   
------------+---------
          1 | vineger
(1 row)
```

However, if we use fuzzy search, a search speed may slow and noises of search result may increase

A search keyword is input to argument of `pgroonga_escape()`.
Also, If we increace `max_distance`, PGroonga search more loose condition.

**Comment by kou** [comment]
ref: #100
